src/com/mendix/widget/GoogleMaps/components/Map.ts

```diff
@@ -1,99 +1,144 @@
-            this.centerToDefault
-        );
-        google.maps.event.addDomListener(window, "resize", () => {
-            const center = this.map.getCenter();
-            google.maps.event.trigger(this.map, "resize");
-            this.map.setCenter(center);
-        });
+    private handleOnGoogleApiLoaded() {
+        this.setState({ isLoaded: true });
+        this.centerMap();
+        this.plotAddresses();
     }
 
-    private getLocation(address: string, successCallback: Function, failureCallback?: Function) {
-        const geocoder = new google.maps.Geocoder();
-        geocoder.geocode({ address }, (results, status) => {
-            if (status === google.maps.GeocoderStatus.OK) {
-                successCallback(results[0].geometry.location);
+    private centerMap() {
+        this.getLocation(this.props.defaultCenterAddress as string, (location: LatLng) => {
+            if (location) {
+                this.setState({ center: location });
             } else {
-                failureCallback();
+                this.setState({ center: this.defaultCenterLocation });
             }
         });
     }
 
-    private createMarker(location: google.maps.LatLng) {
-        Marker({ location, map: this.map });
-        this.map.setCenter(location);
+    private plotAddresses() {
+        this.props.locations.filter((location) => location.address && !location.latitude)
+            .map((locationObject: Location, index) => {
+                this.getLocation(locationObject.address as string, (location: LatLng) => {
+                    if (location) {
+                        locationObject.latitude = Number(location.lat);
+                        locationObject.longitude = Number(location.lng);
+                        this.setState({ locations: this.props.locations });
+                    }
+                });
+            });
     }
 
-    private centerToDefault() {
-        this.getLocation(this.defaultCenter, (defaultLocation: google.maps.LatLng) => {
-            this.map.setCenter(defaultLocation);
-            mx.ui.error(`Could not find location from address ${this.props.address}`);
-        });
+    private getLocation(address: string, callback: (result: LatLng | null) => void) {
+        if (this.state.isLoaded) {
+            const geocoder = new google.maps.Geocoder();
+            geocoder.geocode({ address }, (results, status) => {
+                if (status === google.maps.GeocoderStatus.OK) {
+                    callback({
+                        lat: results[0].geometry.location.lat(),
+                        lng: results[0].geometry.location.lng()
+                    });
+                } else {
+                    this.setState({ alertMessage: `Can not find address ${address}` });
+                    callback(null);
+                }
+            });
+        } else {
+            callback(null);
+        }
+    }
+
+    private createMaker(locationObject: Location, index: number) {
```

Comment:
I have added a turn and i have changed the function to create an array of markers